core/src/labels/labels.cpp

```diff
@@ -169,187 +174,135 @@ void Labels::skipTransitions(const std::vector<std::unique_ptr<Style>>& _styles,
     }
 }
 
-void Labels::checkRepeatGroups(std::vector<Label*>& _visibleSet) const {
+bool Labels::labelComparator(const LabelEntry& _a, const LabelEntry& _b) {
+    if (_a.proxy != _b.proxy) {
+        return _b.proxy;
+    }
+
+    if (_a.priority != _b.priority) {
+        return _a.priority < _b.priority;
+    }
+
+    auto l1 = _a.label;
+    auto l2 = _b.label;
+
+    // Note: This causes non-deterministic placement, i.e. depending on
+    // navigation history.
+    if (l1->occludedLastFrame() != l2->occludedLastFrame()) {
```

Comment:
Is this needed as a sorting rule? So we base the sorting only on things not related to label state.